### DISCUSSION TITLE
Is it possible to provide a custom format for the order id generated by the platform?

### DISCUSSION THREAD

**Discussion by sidd-kulk** [medusajs/medusa#10112] [Q&A]
The current format of the order id is : order_01JCJV7K75MVMF25M1N7658BH5 
But we would like to customize this to another format, say UUID or limit it to a certain length. Is there a way to customize the format for this?  

Thanks in advance.

**Comment by srindom** [comment]
Hi @sidd-kulk, What are you looking to achieve with the customization of the format?

**Reply by sidd-kulk** [reply]
Thanks for your reply. 
It's a very quaint requirement from an arcane downstream system.  That system only allows 25 characters for any field in their 35 year old Database.  And that length cannot be changed.  Hence wanted to generate an order ID of a smaller length.

**Reply by srindom** [reply]
I see - unfortunately, not something we can support. However, maybe the order.display_id could work?

**Reply by sidd-kulk** [reply]
Is display id guaranteed unique?

**Reply by srindom** [reply]
There's actually not a unique constraint on the column; probably should be though. In practice it does become unique however, since it relies on an auto increment in the db.

**Reply by sidd-kulk** [reply]
Super, thanks!
By the way, what is the algorithm used to generate the 32 character order id?  What is the level of randomness and collision probability?

**Reply by srindom** [reply]
It's a ULID id. 

> - 128-bit compatibility with UUID
> - 1.21e+24 unique ULIDs per millisecond
> - Lexicographically sortable!
> - Canonically encoded as a 26 character string, as opposed to the 36 character UUID
> - Uses Crockford's base32 for better efficiency and readability (5 bits per character)
> - Case insensitive
> - No special characters (URL safe)
> - Monotonic sort order (correctly detects and handles the same millisecond)

See more: https://github.com/ulid/spec

**Reply by sidd-kulk** [reply]
Thanks much!  This helps.
